### ISSUE TITLE
plain text mails: unwanted extra line spacing on return

### ISSUE THREAD

**Issue by kolAflash** [RealRaven2000/SmartTemplate4#339]
When SmartTemplate is enabled for replies or forwarding, I get extra vertical spacing between lines when pressing return/enter.

I've configured my Thunderbird to write plain text mails. But this spacing looks like the spacing Thunderbird does when no add ons are installed and writing HTML mails is enabled.

~~Also there's a weird single space at the first line. But I can delete it manually, so the spacing is the bigger problem.~~  
_UPDATE:_  
_Found out, it's the `Insert Space to highlight %cursor%` setting I need to disable, to remove this single white space :-)_  
_I'd suggest to disable this by default, so people don't get unexpected changes when installing SmartTemplate._

&nbsp;

# Details

![grafik](https://github.com/user-attachments/assets/ee9ea627-1717-46d2-a223-99c31e7f1e31)

When sending the mail and looking into my `Send mails` folder, it still contains the extra spacing and the useless space. The mail source contains ` Â ` at the position of the space. (#338 maybe related?)  
```
Content-Type: text/plain; charset=UTF-8; format=flowed
Content-Transfer-Encoding: 8bit

 Â <- strange extra space

weird spacing between lines I type and second line (no empty line in 
between)

third line

On Fri, 22 Nov 2024 11:17:21 +0100, Peter wrote:
> Sehr geehrte Damen und Heren,
> 
> hatten Sie meine E-Mail Ende Oktober erhalten?
```

# Configuration

I've configured my Thunderbird-128 ([Debian-12 Linux, KDE](https://cdimage.debian.org/debian-cd/current-live/amd64/iso-hybrid/)) like this:
- `Account Settings` -> `Attach the signature from a file instead (text, HTML or image)`: `/path/to/signature.txt`
- `Account Settings` -> `Composition & Addressing` -> `Compose messages in HTML format`: `off`
- `Settings` -> `Composition` -> `Sending Format`: `Automatic`
  - tried changing to `Only Plain Text` didn't helped

&nbsp;

I'm using mostly default for the SmartTemplate settings. I've tried changing a few boolean settings, but the issue remains.  
_(I've enabled and disabled each. But I didn't try every possible permutation of settings.)_
- `Account templates` -> `Reply` / `Forward`
  - `Use HTML (e.g., <b> bold </b>)`
    - `Replace line breaks with <br>`
  - `Use instead of default quote header`
- `Global Settings` -> `Replace line breaks in plain text signature with <br>`
- `Advanced Settings` -> `Inject contents of <head> from template`
  - _(beside: you may add a documentation hint, I had to do some searching to find it in #79 and in the [4.1 release notes](https://smarttemplates.quickfolders.org/version.html#4.1))_

`Template` is empty. I only use the `Quote Header` as seen in the screenshot.

&nbsp;

# Expected behavior

Without SmartTemplate installed, Thunderbird puts no extra spacing between lines.
So I'd like Thunderbird to behave the same as without SmartTemplate installed.

&nbsp;

---

Thanks for the great Open Source work!
kolAflash
_(an otherwise happy and paying user)_

**Comment by RealRaven2000** [comment]
> * `Account templates` -> `Reply` / `Forward`
>   
>   * `Use HTML (e.g., <b> bold </b>)`
>     
>     * `Replace line breaks with <br>`
>   * `Use instead of default quote header`
> * `Global Settings` -> `Replace line breaks in plain text signature with <br>`
> * `Advanced Settings` -> `Inject contents of <head> from template`
>   
>   * _(beside: you may add a documentation hint, I had to do some searching to find it in [Inject contents of `<head>` tags into composer's head section. Also overwrite `DOCTYPE` and all `<body>` attributes #79](https://github.com/RealRaven2000/SmartTemplates/issues/79) and in the [4.1 release notes](https://smarttemplates.quickfolders.org/version.html#4.1))_
> 
> `Template` is empty. I only use the `Quote Header` as seen in the screenshot.

Sorry for the late reply - I don't understand what you have checked / unchecked, can you please export your settings:

![Image](https://github.com/user-attachments/assets/901a8487-7193-474d-a85a-a207d43a8b87)

you can upload the json file here on Github or send it to me off list if you prefer. A typical setting for somebody who prefers plain text would be:

![Image](https://github.com/user-attachments/assets/83d261da-130c-45f1-a2af-49c5cb82ebf3)

**Comment by RealRaven2000** [comment]
Also typically, for using composer like a typewriter (just lines followed by more lines, no paragraphs) you would disable the setting Composition >> HTML style >> Use paragraphs format instead of body text by default:

![Image](https://github.com/user-attachments/assets/859b6843-7153-4908-be36-05cfe3660823)

There also is a separate setting in SmartTemplates now that ensures that a real paragraph is inserted., in _Global Settings_ >> _Advanced  Functions_, uncheck _Force Paragraph Mode_.

![Image](https://github.com/user-attachments/assets/2e80e431-9663-4c20-bca5-b912970efbb3)

**Comment by RealRaven2000** [comment]
OK, tested the behavior with "Force Paragraph mode" disabled, here is the result:

![Image](https://github.com/user-attachments/assets/24cab1b2-9d29-4ecb-be7b-d2f5fa8c4fa8)

then we get the "glued together look" of a typewriter, no nice spaces between paragraphs as desired?

Paragraph mode just means that composer functions more like a word processer (think MS word), where you always are in a paragraph unless you select "heading" or  "bullet list". 

`Enter` will always split the paragraph into 2. To get single line breaks (which I barely use) you would have to hit `Shift`+`Enter`.

**Comment by RealRaven2000** [comment]
I will close the issue for now, if you feel the settings above don't work or you find new problems feel free to reopen!

**Comment by kolAflash** [comment]
> OK, tested the behavior with "Force Paragraph mode" disabled [...]

Ah, thanks! Disabling that setting solved the issue.

&nbsp;

Surely I've seen text written with vertical spacing between paragraphs. But I actually never knew, that people call this behavior _"Paragraph mode"_. That's why I extensively described the behavior in my initial message, instead of simply calling it _"Paragraph mode"_.  
> When SmartTemplate is enabled for replies or forwarding, I **get extra vertical spacing between lines when pressing return/enter.**

Even the additional tooltip in the recent [4.10.2pre72](https://github.com/RealRaven2000/SmartTemplates/issues/338#issuecomment-2677125154) doesn't make an explanatory connection between the name _"Paragraph mode"_ and the actual behavior in the UI.

&nbsp;

Anyway, big thanks! 😃

**Comment by RealRaven2000** [comment]
Yes, since this is a question and it didn't cause any code changes, it is ok to leave the issue as closed.

to explain, in the background (even in plain text mode I believe) there are always some HTML entities that format the text, one of these is the paragraph tag:

```html
<p>
  This is a paragraphs. and it ends here:
</p>
```

this is a "block-level" element, which means it usually takes the whole width of the parent container (e.g. the <body> element which contains all visible stuff in an email or web page)

Paragraphs are by default imbued with layout attributes such as block-level margins (space above and below). This make sure that a document written with these has consistent distances between parameters (and not an arbitrary amount of line breaks). Beyond typographical niceness, it's also important for screen readers and data extraction tools to show which semantic portions of information belong together. That's not something that line breaks can do.

The amount of space is determined by the browser (read: mail client) and can be overwritten with css rules to adjust it to your personal preferences. But please never set that to 0, because it effectively makes paragraphs "invisible" to the reading eye and plunges us back into the dark ages of the typewriter.

```html
<style>
  p {
    margin-bottom: 0.3em;
    margin-top: 0.3em;
  }
</style>
```

(unless removed) these rules are global and apply to the complete email, including quoted text and all future replies!

**Comment by RealRaven2000** [comment]
PS: forgot to format the code examples with the backwards apostrophes initially, so they may have looked weird in the original version. I always edit my answers directly on Github, so consumption there is definitely preferred over reading the notification email only.
